nautobot_secrets_providers/providers/hashicorp.py

```diff
@@ -61,26 +68,33 @@ class ParametersForm(BootstrapMixin, forms.Form):
         )
         mount_point = forms.CharField(
             required=False,
-            help_text=f"The path where the secret engine was mounted on (Default: <code>{DEFAULT_MOUNT_POINT}</code>)",
-            initial=DEFAULT_MOUNT_POINT,
+            help_text="The path where the secret engine was mounted on (Default: <code>secret</code>)",
         )
         kv_version = forms.ChoiceField(
             required=False,
-            choices=HashicorpKVVersionChoices,
-            help_text=f"The version of the kv engine (either v1 or v2) (Default: <code>{DEFAULT_KV_VERSION}</code>)",
-            initial=DEFAULT_KV_VERSION,
+            choices=add_blank_choice(HashicorpKVVersionChoices),
+            help_text="The version of the kv engine (either v1 or v2) (Default: <code>v2</code>)",
+            label="KV Version",
+        )
+        hashicorp_config = forms.ChoiceField(
+            required=False,
+            choices=hashicorp_config_choices,
+            help_text="",
         )
 
     @classmethod
-    def validate_vault_settings(cls, secret=None):
+    def validate_vault_settings(cls, secret=None, vault_config=None):
         """Validate the vault settings."""
+        vault_settings = settings.PLUGINS_CONFIG["nautobot_secrets_providers"].get("hashicorp_vault", {})
+        if vault_config and vault_config != "default":
+            vault_settings = settings.PLUGINS_CONFIG["nautobot_secrets_providers"]["hashicorp_vault"][
+                "configurations"
+            ].get(vault_config, {})
         # This is only required for HashiCorp Vault therefore not defined in
         # `required_settings` for the plugin config.
-        plugin_settings = settings.PLUGINS_CONFIG["nautobot_secrets_providers"]
-        if "hashicorp_vault" not in plugin_settings:
+        if not vault_settings:
             raise exceptions.SecretProviderError(secret, cls, "HashiCorp Vault is not configured!")
```

Comment:
```suggestion
            if vault_config is None:
                 vault_config = "default"
            raise exceptions.SecretProviderError(secret, cls, f"HashiCorp Vault {vault_config} is not configured!")

```

Reply:
I like this, but then it will likely go above the branching limit, I will apply and see if it breaks.